### ISSUE TITLE
Python error in fresh macOS Sonoma install with Synfig 1.5.3

### ISSUE THREAD

**Issue by pgilfernandez** [synfig/synfig#3420]
**Describe the bug**
When trying to use any Python related commands (plugins or export to Lottie...) Synfig says the following:
![Captura de pantalla 2024-09-09 a las 9 51 32](https://github.com/user-attachments/assets/bb19a01c-b8ed-463b-ac36-ac9d9cfd7536)

**To Reproduce**
1. get a mac
2. install a fresh Sonoma OS
3. install Synfig v1.5.3
4. open it (it works)
5. try to run a Plugin

**Expected behavior**
Python and Synfig are correctly setup and the plugin works

**System information:**
 - OS: Mac
 - OS version: macOS Sonoma 14.4.1
 - Synfig version: 1.5.3

**Additional comments**
By default, Sonoma has Python 3.9.6 installed at `/usr/bin/python3` (at least that's what `which python3` throws...)
Thanks!

**Comment by rodolforg** [comment]
Python version is checked here:
https://github.com/synfig/synfig/blob/02b4d0835638b6564d1192a2cc930fd4ff466ea7/synfig-studio/src/gui/app.cpp#L4107-L4130

Python path is retrieved here:
https://github.com/synfig/synfig/blob/02b4d0835638b6564d1192a2cc930fd4ff466ea7/synfig-studio/src/gui/pluginmanager.cpp#L465-L518

**Comment by pgilfernandez** [comment]
> Python version is checked here:
> 
> https://github.com/synfig/synfig/blob/02b4d0835638b6564d1192a2cc930fd4ff466ea7/synfig-studio/src/gui/app.cpp#L4107-L4130
> 
> 
> 
> Python path is retrieved here:
> 
> https://github.com/synfig/synfig/blob/02b4d0835638b6564d1192a2cc930fd4ff466ea7/synfig-studio/src/gui/pluginmanager.cpp#L465-L518
> 
> 
> 
> 

Do you want me to do anything with them? Any test or check?

**Comment by pgilfernandez** [comment]
ping @rodolforg
;)

**Comment by rodolforg** [comment]
Hey @pgilfernandez  :)
I just listed what I know about where Synfig tries to find the python binary.

1. What is the full path of the Python 3.9.x installed on your OS?
2. try to force to use it by setting the environment variable SYNFIG_PYTHON_BINARY to it
https://github.com/synfig/synfig/blob/02b4d0835638b6564d1192a2cc930fd4ff466ea7/synfig-studio/src/gui/pluginmanager.cpp#L474

**Comment by pgilfernandez** [comment]
Hi @rodolforg,

OK, but, in order to set the environment variable SYNFIG_PYTHON_BINARY to the default macOS one I would need to compile Synfig, right?

**Comment by pgilfernandez** [comment]
finally I solved it by manually downloading and installing Python 3.7 but, new users won't be able to figure this out... Synfig looks to be forced to use an specific version and not, for instance, 3.7 or higher. I had 3.9 installed by default and it didn't work... what to you think?

**Comment by rodolforg** [comment]
> in order to set the environment variable SYNFIG_PYTHON_BINARY to the default macOS one I would need to compile Synfig, right?

No. https://support.apple.com/guide/terminal/apd382cc5fa-4f58-4449-b20a-41c53c006f8f/mac

> Synfig looks to be forced to use an specific version and not, for instance, 3.7 or higher.

The code I showed does not force a version. It tries to find "python3" or "python" binary in `PATH`.

**Comment by rodolforg** [comment]
@ice0 @morevnaproject
